### PR TITLE
Add support for moving folders to system trash

### PR DESCRIPTION
**Pull request by Fleker**
Adds a `--trash` flag to the CLI to move a `node_modules` to system trash rather than deleting directly.

To use the `trash` module, the project needs to be rewritten to use ESModules
There is one test still outstanding that isn't running as I work out some compat issues, but wanted to create this now to solicit some feedback.

See #60

### CODE REVIEW

File: src/constants/index.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
...
-export * from './spinner.constants';
-export * from './update.constants';
-export * from './recursive-rmdir-node-support.constants';
+export * from './cli.constants.js';
+export * from './main.constants.js';
+export * from './messages.constants.js';
+export * from './sort.result.js';
+export * from './spinner.constants.js';
+export * from './update.constants.js';
+export * from './recursive-rmdir-node-support.constants.js';
```
Comment (mrbaguvix) [inline src/constants/index.ts:7]: Just curious why suffix file names with `.js` extension?

File: src/services/index.ts [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
-export * from './windows-files.service';
+export * from './console.service.js';
+export * from './files.service.js';
+export * from './https.service.js';
+export * from './linux-files.service.js';
+export * from './results.service.js';
+export * from './spinner.service.js';
+export * from './stream.service.js';
+export * from './update.service.js';
+export * from './windows-files.service.js';
```
Comment (mrbaguvix) [inline src/services/index.ts:9]: huh?

**Review by mrbaguvix — CHANGES_REQUESTED** [review]
Nice one adding the `--trash` option but if adding one single feature could make rewriting most of the original project structure, i don't think it is ideal for a merge. You can look for a typescript port for `trash` lib or an alternative package.

**Comment by zaldih** [comment]
Hi!
It's been a while since I had to decide what to do whith this precious PR. The problem is that initially I didn't really understand why it was necessary to modify so many things just for the feature, so one of the tasks I thought to do was to "upate" the other lib to make it compatible with projects like this. Actually the one who was wrong was me since the future points to ESM although eve today it is still experimental.
That is why I finally decided to update the project with ESM (already in develop branch).

Now that there should be no more ESM problems, **could you reopen the PR with only the changes related to the feature?** I will merge it as soon as possible. Thank!

**Pull request closed without merging**